javaparser-core/src/main/java/com/github/javaparser/Providers.java

```diff
@@ -57,4 +57,27 @@ public static Provider provider(String source) {
         return new StringProvider(assertNotNull(source));
     }
 
+
+
+    /**
+     * Provide a Provider from the resource found in classLoader
+     * @param classLoader Class loader in which resource is searched
+     * @param pathToResource path to resource in provided class loader. As working relatively to a class loader, no "/"
+     * at the beginning of the path
+     * @param encoding encoding
+     * @return the provider associated to pathToResource
+     */
+    public static Provider resourceProvider(ClassLoader classLoader, String pathToResource, Charset encoding) {
+        return provider(classLoader.getResourceAsStream(pathToResource), encoding);
+    }
+
+    public static Provider resourceProvider(String pathToResource, Charset encoding) {
```

Comment:
I'm far to be a master of clarity :) 
I will add javadoc to each of the methods. In order to add the precision for each one of them.
Is the first javadoc OK for you?